sklearn/model_selection/_prediction.py

```diff
@@ -0,0 +1,462 @@
+        * `"tpr"`: find the decision threshold for a true positive ratio (TPR)
+          of `objective_value`;
+        * `"tnr"`: find the decision threshold for a true negative ratio (TNR)
+          of `objective_value`.
+
+    objective_metric_params : dict, default=None
+        Some extra parameters to pass to `objective_metric`.
+
+    objective_value : float, default=None
+        The value associated with the `objective_metric` metric for which we
+        want to find the decision threshold when `objective_metric` is equal to
+        `"tpr"` or `"tnr"`.
+
+    method : {"auto", "decision_function", "predict_proba"}, default="auto"
+        Methods by the classifier `base_estimator` corresponding to the
+        decision function for which we want to find a threshold. It can be:
+
+        * if `"auto"`, it will try to invoke, for each classifier,
+          `"decision_function` or `"predict_proba"` in that order.
+        * otherwise, one of `"predict_proba"` or `"decision_function"`.
+          If the method is not implemented by the classifier, it will raise an
+          error.
+
+    n_threshold : int, default=1000
+        The number of decision threshold to use when discretizing the output
+        of the classifier `method`.
+
+    pos_label : int or str, default=None
+        The label of the positive class. When `pos_label=None`, if `y_true` is
+        in `{-1, 1}` or `{0, 1}`, `pos_label` is set to 1, otherwise an error
+        will be raised.
+
+    cv : int, float, cross-validation generator, iterable or "prefit", \
+            default=None
+        Determines the cross-validation splitting strategy used in
+        `cross_val_predict` to train classifier. Possible inputs for cv are:
+
+        * None, to use the default 5-fold stratified K-fold cross validation;
+        * An integer number, to specify the number of folds in a stratified
+          k-fold;
+        * A float number, to specify a single shuffle split. The floating
+          number should be in (0, 1) and represent the size of the validation
+          set;
+        * An object to be used as a cross-validation generator;
+        * An iterable yielding train, test splits;
+        * "prefit", to bypass the cross-validation.
+
+        Refer :ref:`User Guide <cross_validation>` for the various
+        cross-validation strategies that can be used here.
+
+    refit : "auto" or bool, default="auto"
+        Whether or not to refit the classifier on the entire training set once
+        the decision threshold has been found. By default, `refit="auto"` is
+        equivalent to `refit=False` when `cv` is a float number using a single
+        shuffle split or `cv="prefit"` otherwise `refit=True` in all other
+        cases. Note that forcing `refit=False` on cross-validation having more
+        than a single split will raise an error. Similarly, `refit=True` in
+        conjunction with `cv="prefit"` will raise an error.
```

Comment:
I understand the logic, but is it overkill? I feel like this should follow the same simple logic as the Grid Search estimators

Reply:
If we choose the way of the GridSearchCV, `refit=False` will raise most of the time an error.
If we choose `refit=True`, it will refit when a user ask for a single split while it would be the case where you want to be efficient most probably (because you have a large dataset).
So the "auto" switch from one to another and seem a good default to me. If I recall, we made this choice with @ogrisel